src/components/CharactersPage/CharacterForm/CharacterForm.js

```diff
@@ -107,6 +108,19 @@ const CharacterForm = ({ onClose }) => {
     reValidateMode: "onSubmit",
   })
 
+  useEffect(() => {
+    if (isEdit) {
+      const cachedCharacters = client.cache.extract()
```

Comment:
You should defiantly not be doing this. The `extract` method returns a normalised version of everything in the cache, and you certainly shouldn't be reading everything in the cache. Also, what happens if a character is removed while the form is open?